super-intel.c

```diff
@@ -2499,11 +2499,11 @@ static int ahci_enumerate_ports(struct sys_dev *hba, int port_count, int host_ba
 	if (dir)
 		closedir(dir);
 	if (err == 0) {
-		int i;
+		int64_t i;
 
 		for (i = 0; i < port_count; i++)
-			if (port_mask & (1 << i))
-				printf("          Port%d : - no device attached -\n", i);
+			if (port_mask & (1LL << i))
```

Comment:
port_mask is `unsigned long` so `i` should be `unsigned long` and respectively proper specifier should be used for printing (probably %ld is right).

Reply:
changing i to unsigned long causes an error:

super-intel.c: In function ‘ahci_enumerate_ports’:
super-intel.c:2504:31: error: comparison of integer expressions of different signedness: ‘long unsigned int’ and ‘int’ [-Werror=sign-compare]
 2504 |                 for (i = 0; i < port_count; i++)
      |                               ^
cc1: all warnings being treated as errors

Reply:
I see, thanks.
I don't like the approach with `int64_t` because it is strong typing. We are forcing `i` to be 64 and that may cause the issue on different architectures and is generally less readable.
IMHO the best is to ensure that i and port_count are same. I did following changes and they are working for me:

```diff
diff --git a/super-intel.c b/super-intel.c
index 6314ec3e1e86..407d1263a95e 100644
--- a/super-intel.c
+++ b/super-intel.c
@@ -2343,7 +2343,8 @@ void print_encryption_information(int disk_fd, enum sys_dev_type hba_type)
               get_encryption_status_string(information.status));
 }

-static int ahci_enumerate_ports(struct sys_dev *hba, int port_count, int host_base, int verbose)
+static int ahci_enumerate_ports(struct sys_dev *hba, unsigned long port_count, int host_base,
+                                int verbose)
 {
        /* dump an unsorted list of devices attached to AHCI Intel storage
         * controller, as well as non-connected ports
@@ -2357,7 +2358,7 @@ static int ahci_enumerate_ports(struct sys_dev *hba, int port_count, int host_ba

        if (port_count > (int)sizeof(port_mask) * 8) {
                if (verbose > 0)
-                       pr_err("port_count %d out of range\n", port_count);
+                       pr_err("port_count %ld out of range\n", port_count);
                return 2;
        }

@@ -2499,7 +2500,7 @@ static int ahci_enumerate_ports(struct sys_dev *hba, int port_count, int host_ba
        if (dir)
                closedir(dir);
        if (err == 0) {
-               int64_t i;
+               unsigned long i;

                for (i = 0; i < port_count; i++)
                        if (port_mask & (1LL << i))

```

Could you please test it?

Reply:
You forgot the %ld on the printf....
I also see you have an "if (port_mask & (1LL << i))"   which is not in your diff.

```
super-intel.c: In function ‘ahci_enumerate_ports’:
super-intel.c:2507:56: error: format ‘%d’ expects argument of type ‘int’, but argument 2 has type ‘long unsigned int’ [-Werror=format=]
 2507 |                                 printf("          Port%d : - no device attached -\n", i);
      |                                                       ~^                              ~
      |                                                        |                              |
      |                                                        int                            long unsigned int
      |                                                       %ld
cc1: all warnings being treated as errors

```

Reply:
This worked for me....

`diff --git a/super-intel.c b/super-intel.c
index 2b8b6fda..bac67b93 100644
--- a/super-intel.c
+++ b/super-intel.c
@@ -2343,7 +2343,8 @@ void print_encryption_information(int disk_fd, enum sys_dev_type hba_type)
 	       get_encryption_status_string(information.status));
 }
 
-static int ahci_enumerate_ports(struct sys_dev *hba, int port_count, int host_base, int verbose)
+static int ahci_enumerate_ports(struct sys_dev *hba, unsigned long port_count, int host_base,
+				int verbose)
 {
 	/* dump an unsorted list of devices attached to AHCI Intel storage
 	 * controller, as well as non-connected ports
@@ -2357,7 +2358,7 @@ static int ahci_enumerate_ports(struct sys_dev *hba, int port_count, int host_ba
 
 	if (port_count > (int)sizeof(port_mask) * 8) {
 		if (verbose > 0)
-			pr_err("port_count %d out of range\n", port_count);
+			pr_err("port_count %ld out of range\n", port_count);
 		return 2;
 	}
 
@@ -2499,11 +2500,11 @@ static int ahci_enumerate_ports(struct sys_dev *hba, int port_count, int host_ba
 	if (dir)
 		closedir(dir);
 	if (err == 0) {
-		int i;
+		unsigned long i;
 
 		for (i = 0; i < port_count; i++)
 			if (port_mask & (1 << i))
-				printf("          Port%d : - no device attached -\n", i);
+				printf("          Port%ld : - no device attached -\n", i);
 	}
 
 	return err;
-- 
2.31.1
`

Reply:
diff --git a/super-intel.c b/super-intel.c
index 2b8b6fda..bac67b93 100644
--- a/super-intel.c
+++ b/super-intel.c
@@ -2343,7 +2343,8 @@ void print_encryption_information(int disk_fd, enum sys_dev_type hba_type)
 	       get_encryption_status_string(information.status));
 }
 
-static int ahci_enumerate_ports(struct sys_dev *hba, int port_count, int host_base, int verbose)
+static int ahci_enumerate_ports(struct sys_dev *hba, unsigned long port_count, int host_base,
+				int verbose)
 {
 	/* dump an unsorted list of devices attached to AHCI Intel storage
 	 * controller, as well as non-connected ports
@@ -2357,7 +2358,7 @@ static int ahci_enumerate_ports(struct sys_dev *hba, int port_count, int host_ba
 
 	if (port_count > (int)sizeof(port_mask) * 8) {
 		if (verbose > 0)
-			pr_err("port_count %d out of range\n", port_count);
+			pr_err("port_count %ld out of range\n", port_count);
 		return 2;
 	}
 
@@ -2499,11 +2500,11 @@ static int ahci_enumerate_ports(struct sys_dev *hba, int port_count, int host_ba
 	if (dir)
 		closedir(dir);
 	if (err == 0) {
-		int i;
+		unsigned long i;
 
 		for (i = 0; i < port_count; i++)
 			if (port_mask & (1 << i))
-				printf("          Port%d : - no device attached -\n", i);
+				printf("          Port%ld : - no device attached -\n", i);
 	}
 
 	return err;
-- 
2.31.1

Reply:
---
 super-intel.c | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/super-intel.c b/super-intel.c
index 2b8b6fda..bac67b93 100644
--- a/super-intel.c
+++ b/super-intel.c
@@ -2343,7 +2343,8 @@ void print_encryption_information(int disk_fd, enum sys_dev_type hba_type)
 	       get_encryption_status_string(information.status));
 }
 
-static int ahci_enumerate_ports(struct sys_dev *hba, int port_count, int host_base, int verbose)
+static int ahci_enumerate_ports(struct sys_dev *hba, unsigned long port_count, int host_base,
+				int verbose)
 {
 	/* dump an unsorted list of devices attached to AHCI Intel storage
 	 * controller, as well as non-connected ports
@@ -2357,7 +2358,7 @@ static int ahci_enumerate_ports(struct sys_dev *hba, int port_count, int host_ba
 
 	if (port_count > (int)sizeof(port_mask) * 8) {
 		if (verbose > 0)
-			pr_err("port_count %d out of range\n", port_count);
+			pr_err("port_count %ld out of range\n", port_count);
 		return 2;
 	}
 
@@ -2499,11 +2500,11 @@ static int ahci_enumerate_ports(struct sys_dev *hba, int port_count, int host_ba
 	if (dir)
 		closedir(dir);
 	if (err == 0) {
-		int i;
+		unsigned long i;
 
 		for (i = 0; i < port_count; i++)
 			if (port_mask & (1 << i))
-				printf("          Port%d : - no device attached -\n", i);
+				printf("          Port%ld : - no device attached -\n", i);
 	}
 
 	return err;
-- 
2.31.1

Reply:
My formatting above didn't come out right... but it just has the additional fix to the last printf with %ld.

Reply:
@ncroxon I made changes of top of your patch, maybe that is why I omitted some places.
Anyway, LGTM.